Block Editor: Creating Blocks

chynu opened the pull request:
Can define blocks. It will update the preview, block definition, and the generator stub.

Currently, testing for these features will not work because it depends on some functions in `ProjectController` and some model classes (`BlockLibrarySet`, `BlockLibrary`, etc.) which are not yet implemented (but will be in upcoming PR's).

**EDIT:** Code will break when testing because I have now removed functions in `ProjectController` that have already been implemented in #118 (to avoid merge conflicts).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-devtools/117)
<!-- Reviewable:end -->

chynu committed:
Updating previews on workspace change

chynu committed:
Updating the navtree upon changes

chynu committed:
Changes wrt ProjectController in updating tree

chynu committed:
Implemented some controller-side functions.

chynu committed:
Reverting irrelevant changes.

svouse commented:
Review status: 0 of 11 files reviewed at latest revision, 13 unresolved discussions.

---

*[src/controller/block_editor_controller.js, line 42 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRs4MRzd8KLCYqE7fb:-KpRs4MRzd8KLCYqE7fc:b9goir9) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L42)):*
> ```JavaScript
>   /**
>    * @constructor
>    * @param {!Project} project Project object associated with this controller.
> ```

Needs to be updated to ProjectController

---

*[src/controller/block_editor_controller.js, line 46 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRs1cEHOkIIFFxh9AP:-KpRs1cEHOkIIFFxh9AQ:bgcp112) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L46)):*
> ```JavaScript
>    *     used to generate Blockly objects for import/export.
>    * @param {!NavigationTree} navTree Navigation tree that will be updated after
>    *     changes in this editor.
> ```

This param isn't passed in

---

*[src/controller/block_editor_controller.js, line 104 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRsQRtrosm3AbG6XvC:-KpRsQRtrosm3AbG6XvD:bxlv16l) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L104)):*
> ```JavaScript
>   /**
>    * Refreshes all three previews (block preview, block definition view, and
>    * generator stub) at once. Also updates the model side (BlockDefinition
> ```

Nit: consider changing to "Updates BlockDefinition", or something more concise 

---

*[src/controller/block_editor_controller.js, line 132 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRsffepcvB5WnH_iLv:-KpRsffepcvB5WnH_iLw:b-en8t69) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L132)):*
> ```JavaScript
> 
>   /**
>    * Updates model side of block definition. Updates the type field and the
> ```

Similarly, "Updates blockType and XML of BlockDefinition"

---

*[src/controller/block_editor_controller.js, line 157 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRt3K-Edky4471MLNN:-KpRt3K-Edky4471MLNO:b-sxpngp) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L157)):*
> ```JavaScript
>   /**
>    * Updates the Block Definition textarea with proper JSON or JavaScript.
>    * @param {string} format Format of block definition. Either 'JSON' or 'JavaScript'.
> ```

@private?

---

*[src/controller/block_editor_controller.js, line 313 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtDZ26Qc_yjT3wCg9:-KpRtDZ26Qc_yjT3wCgA:b9galu6) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L313)):*
> ```JavaScript
> 
>   /**
>    * Gets the block type name that is currently being edited. Extracts type from
> ```

Nit: just "block type" instead of "block type name"

---

*[src/controller/editor_controller.js, line 46 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtTOrrXw3YNQAIVUw:-KpRtTOrrXw3YNQAIVUx:bnzjldw) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/editor_controller.js#L46)):*
> ```JavaScript
>   constructor(projectController, hiddenWorkspace) {
>     /**
>      * ProjectController which controls changes to this Project.
> ```

Nit: could you change "this" to "the"? I almost read it as a field 

---

*[src/controller/project_controller.js, line 91 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtrzgnHZsPpE786H4:-KpRtrzgnHZsPpE786H5:b-vkn8l2) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L91)):*
> ```JavaScript
>    */
>   createBlockDefinition(blockType, libraryName) {
>     const library = this.project.getLibrary(libraryName);
> ```

exists as of 118 but fine//can be resolved

---

*[src/controller/project_controller.js, line 163 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtwwHqhOdaP1V_YPi:-KpRtwwIqBK0N2sRE01M:btvc76a) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L163)):*
> ```JavaScript
>    */
>   addBlockDefinition(block, libraryName) {
>     const library = this.project.getLibrary(libraryName);
> ```

Same as above. Additionally, JSDoc for block should have type {BlockDefinition} instead of string

---

*[src/controller/project_controller.js, line 255 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuT9mlnkgiUb2u1IF:-KpRuT9mlnkgiUb2u1IG:b-unfrcf) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L255)):*
> ```JavaScript
>    * Renames a block definition.
>    * @param {string} oldName Original type name of BlockDefintion object to change.
>    * @param {string} newName New type name of BlockDefinition object.
> ```

Also in 118

---

*[src/controller/toolbox_controller.js, line 35 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRujwvKTzyahPe-ia3:-KpRujwvKTzyahPe-ia4:b-x1lsux) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/toolbox_controller.js#L35)):*
> ```JavaScript
>  */
> class ToolboxController {
>   constructor(projectController, hiddenWorkspace) {
> ```

Needs JSDoc

---

*[src/controller/toolbox_controller.js, line 311 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuo8I0fDkKqmRhThW:-KpRuo8I0fDkKqmRhThX:b9lvv6j) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/toolbox_controller.js#L311)):*
> ```JavaScript
>   updateEditorToolbox() {
>     const newToolboxXml = FactoryUtils.updateBlockLibCategory(
>         this.projectController.getProject(), this.hiddenWorkspace);
> ```

Does the toolbox need the controller for any reason other than getting the project? Would it make more sense to pass the project, then?

---

*[src/controller/workspace_controller.js, line 150 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuxg0_dGnmvngB82m:-KpRuxg0_dGnmvngB82n:b-ya81ng) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/workspace_controller.js#L150)):*
> ```JavaScript
>   updateEditorToolbox() {
>     const newToolboxXml = FactoryUtils.updateBlockLibCategory(
>         this.projectController.getProject(), this.hiddenWorkspace);
> ```

Same as above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117)*
<!-- Sent from Reviewable.io -->

chynu committed:
Nits for #117

chynu commented:
Review status: 0 of 11 files reviewed at latest revision, 13 unresolved discussions.

---

*[src/controller/block_editor_controller.js, line 42 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRs4MRzd8KLCYqE7fb:-KpS2ocd_hJblW70X-H_:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L42)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Needs to be updated to ProjectController
</blockquote></details>

Done.

---

*[src/controller/block_editor_controller.js, line 46 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRs1cEHOkIIFFxh9AP:-KpS2rBdBJCH_37dIObk:brpmqc3) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L46)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

This param isn't passed in
</blockquote></details>

removed.

---

*[src/controller/block_editor_controller.js, line 104 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRsQRtrosm3AbG6XvC:-KpS2yyjufydRFat4Zle:b-q6ji97) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L104)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Nit: consider changing to "Updates BlockDefinition", or something more concise 
</blockquote></details>

Done (?).

---

*[src/controller/block_editor_controller.js, line 132 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRsffepcvB5WnH_iLv:-KpS3TX0cqJBUz30N_gu:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L132)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Similarly, "Updates blockType and XML of BlockDefinition"
</blockquote></details>

Done.

---

*[src/controller/block_editor_controller.js, line 157 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRt3K-Edky4471MLNN:-KpS3WiYNZnTTjyGLo4n:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L157)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

@private?
</blockquote></details>

Done.

---

*[src/controller/block_editor_controller.js, line 313 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtDZ26Qc_yjT3wCg9:-KpS3_uhxLiXDMk0xUZP:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/block_editor_controller.js#L313)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Nit: just "block type" instead of "block type name"
</blockquote></details>

Done.

---

*[src/controller/editor_controller.js, line 46 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtTOrrXw3YNQAIVUw:-KpS3dO9dRIzSewyxl8t:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/editor_controller.js#L46)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Nit: could you change "this" to "the"? I almost read it as a field 
</blockquote></details>

Done.

---

*[src/controller/project_controller.js, line 91 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtrzgnHZsPpE786H4:-KpS2jwRyBZ35qeqOZkV:b-snjrum) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L91)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

exists as of 118 but fine//can be resolved
</blockquote></details>

Removed function to avoid merge conflicts.

---

*[src/controller/project_controller.js, line 163 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRtwwHqhOdaP1V_YPi:-KpS2lZjvISLJXYuf8Zz:b-snjrum) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L163)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Same as above. Additionally, JSDoc for block should have type {BlockDefinition} instead of string
</blockquote></details>

Removed function to avoid merge conflicts.

---

*[src/controller/project_controller.js, line 255 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuT9mlnkgiUb2u1IF:-KpS2mGVhMLefq9JGr5I:b-snjrum) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/project_controller.js#L255)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Also in 118
</blockquote></details>

Removed function to avoid merge conflicts.

---

*[src/controller/toolbox_controller.js, line 35 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRujwvKTzyahPe-ia3:-KpS2nC18M72fhcT_Kxg:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/toolbox_controller.js#L35)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Needs JSDoc
</blockquote></details>

Done.

---

*[src/controller/toolbox_controller.js, line 311 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuo8I0fDkKqmRhThW:-KpS1hjX2zPzfsLOMM_7:b-bir5eq) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/toolbox_controller.js#L311)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Does the toolbox need the controller for any reason other than getting the project? Would it make more sense to pass the project, then?
</blockquote></details>

The toolbox and workspace editor haven't yet been fully implemented, but the project controller will be necessary for renaming toolboxes, etc. when making changes to the project.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117)*
<!-- Sent from Reviewable.io -->

chynu committed:
Removed functions already implemented in #118

rachel-fenichel commented:
Reviewed 8 of 11 files at r1, 2 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[src/factory_utils.js, line 47 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS4by89cjIXV7LODWN:-KpS4by89cjIXV7LODWO:b22lvlt) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/factory_utils.js#L47)):*
> ```JavaScript
>   const rootBlock = FactoryUtils.getRootBlock(workspace);
>   const blockType = FactoryUtils.cleanBlockType(rootBlock.getFieldValue('NAME'));
>   console.log(blockType);
> ```

Remove log.

---

*[src/controller/block_editor_controller.js, line 129 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5AXuaDY24voy4mec:-KpS5AXuaDY24voy4med:b-12icpm) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/block_editor_controller.js#L129)):*
> ```JavaScript
> 
>   /**
>    * Updates blockType and XML of BlockDefinition.
> ```

Of which BlockDefinition?

---

*[src/controller/block_editor_controller.js, line 253 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5IsSZsVVVqjA09Rx:-KpS5IsSZsVVVqjA09Ry:b29gj9r) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/block_editor_controller.js#L253)):*
> ```JavaScript
> 
>     if (format == BlockEditorController.FORMAT_JSON) {
>       console.log('json format detected!');
> ```

Remove logs (unless you have a good reason to keep them to the next PR)

---

*[src/view/navigation_tree.js, line 231 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5gmBfTQdWUbfs_2X:-KpS5gmBfTQdWUbfs_2Y:b-ethd6y) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/view/navigation_tree.js#L231)):*
> ```JavaScript
>    */
>   deleteBlockLibraryNode(blockLibraryName) {
>     this.deleteComponentNode(LIBRARY_PREFIX, blockLibraryName);
> ```

You two are going to conflict in this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117)*
<!-- Sent from Reviewable.io -->

svouse commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[src/controller/editor_controller.js, line 52 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS8x0PR_jt5buIjCej:-KpS8x0QSeU9s92E7L3i:bp44n4n) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L52)):*
> ```JavaScript
> 
>     /**
>      * Block Editor Controller
> ```

Maybe update the JSDoc to say more about what the field does? We know it's a BlockEditorController from the @type

---

*[src/controller/editor_controller.js, line 59 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS963hzdqM8CWemc5C:-KpS963iuVDUYhnb5_3Q:b-ya81ng) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L59)):*
> ```JavaScript
> 
>     /**
>      * Toolbox Controller.
> ```

Same as above

---

*[src/controller/editor_controller.js, line 66 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS97UWYmzsE8Gnyizx:-KpS97UWYmzsE8Gnyizy:b-ya81ng) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L66)):*
> ```JavaScript
> 
>     /**
>      * Workspace Controller.
> ```

Same as above

---

*[src/controller/toolbox_controller.js, line 311 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuo8I0fDkKqmRhThW:-KpS9Bl-AUnaJU_DwaNH:br26px) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/toolbox_controller.js#L311)):*
<details><summary><i>Previously, chynu (celine choo) wrote…</i></summary><blockquote>

The toolbox and workspace editor haven't yet been fully implemented, but the project controller will be necessary for renaming toolboxes, etc. when making changes to the project.
</blockquote></details>

Cool cool I dig it

---

*[src/controller/workspace_controller.js, line 150 at r1](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpRuxg0_dGnmvngB82m:-KpS9GVosji2q7S4KdOl:bmvauzm) ([raw file](https://github.com/google/blockly-devtools/blob/4a01873b8a8f1bd59747bd890e4048b31dc407e1/src/controller/workspace_controller.js#L150)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Resolved as per earlier comment//disregard this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117)*
<!-- Sent from Reviewable.io -->

svouse commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> after comments

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117#-:-KpS9XGJ-wR5aqP_gvgs:b-lquxi1)*
<!-- Sent from Reviewable.io -->

chynu committed:
Responses to #117

chynu commented:
Review status: 7 of 10 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[src/factory_utils.js, line 47 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS4by89cjIXV7LODWN:-KpS8yUz5gFpkSEc3yCu:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/factory_utils.js#L47)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Remove log.
</blockquote></details>

Done.

---

*[src/controller/block_editor_controller.js, line 129 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5AXuaDY24voy4mec:-KpS98ppl4RNABwsO4pd:b-hahvg8) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/block_editor_controller.js#L129)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Of which BlockDefinition?
</blockquote></details>

The currently open one -- I updated the jsdoc.

---

*[src/controller/block_editor_controller.js, line 253 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5IsSZsVVVqjA09Rx:-KpS9EG7rsa_OcX6Alof:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/block_editor_controller.js#L253)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Remove logs (unless you have a good reason to keep them to the next PR)
</blockquote></details>

Done.

---

*[src/controller/editor_controller.js, line 52 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS8x0PR_jt5buIjCej:-KpS9sx06ct6YNgQfNh2:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L52)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Maybe update the JSDoc to say more about what the field does? We know it's a BlockEditorController from the @type
</blockquote></details>

Done.

---

*[src/controller/editor_controller.js, line 59 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS963hzdqM8CWemc5C:-KpS9tPNbJuTj8mwQq0c:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L59)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Done.

---

*[src/controller/editor_controller.js, line 66 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS97UWYmzsE8Gnyizx:-KpS9tivoingxzn68R8-:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/controller/editor_controller.js#L66)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Same as above
</blockquote></details>

Done.

---

*[src/view/navigation_tree.js, line 231 at r3](https://reviewable.io:443/reviews/google/blockly-devtools/117#-KpS5gmBfTQdWUbfs_2X:-KpS9RUpCgEQUtoFnGgf:b-hixtdf) ([raw file](https://github.com/google/blockly-devtools/blob/b0c8d84fe0a2041ede1ea339d078437d9a64c5f2/src/view/navigation_tree.js#L231)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

You two are going to conflict in this file.
</blockquote></details>

I believe I have now removed functions that Sage has already implemented in #118.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117)*
<!-- Sent from Reviewable.io -->

rachel-fenichel commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/117#-:-KpSAc3HXNvxQooDEPxW:bnfp4nl)*
<!-- Sent from Reviewable.io -->

chynu committed:
Merge branch 'develop' into blockdef
